### ISSUE TITLE
Dense layer without activation?

### ISSUE THREAD

**Issue by owen8877** [pvitoria/ChromaGAN#8]
In
https://github.com/pvitoria/ChromaGAN/blob/30e975beccf1ab94f5c26b0ec3df7b34f22cb612/SOURCE/ChromaGAN.py#L209
there is no activation parameter designated in the Dense layer. In keras 2.3.1, the default activation is linear activation (i.e. no activation):
```
tf.keras.layers.Dense(
    units,
    activation=None,
    ...
```

**Comment by motionlife** [comment]
> In
> https://github.com/pvitoria/ChromaGAN/blob/30e975beccf1ab94f5c26b0ec3df7b34f22cb612/SOURCE/ChromaGAN.py#L209
> 
> 
> there is no activation parameter designated in the Dense layer. In keras 2.3.1, the default activation is linear activation (i.e. no activation):
> ```
> tf.keras.layers.Dense(
>     units,
>     activation=None,
>     ...
> ```

Just want to send the class semantics for colorization head(a.k.a the encoder), so only logits is enough, probably more stable than a softmax result.

**Comment by owen8877** [comment]
But it makes no sense to have multiple fully-connected hidden layers without activation - that is equivalent to (or potentially less than) just one fully-connected layer.

**Comment by motionlife** [comment]
@owen8877 yes, your are right, just take another look at the code, even the classification head didn't have activation, consider that orignal vgg has relu for both the 4096-dense layers.

**Comment by owen8877** [comment]
@motionlife That's true. I hope they can fix the mistake and probably yield better performance.

**Comment by motionlife** [comment]
@owen8877 Or just use one dense layer to make model smaller

**Comment by owen8877** [comment]
@motionlife Well, we might as well stick to the vanilla VGG16 design since there is a classification loss against the pre-trained VGG16 model.

https://github.com/pvitoria/ChromaGAN/blob/30e975beccf1ab94f5c26b0ec3df7b34f22cb612/SOURCE/ChromaGAN.py#L149

I doubt there might be a performance recession if we cut the FC layers thin.

**Comment by motionlife** [comment]
@motionlife yes understand, I mean if there is no performance boost when add back activation. As you said, why not just use one dense.
